pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java

```diff
@@ -4109,6 +4109,35 @@ public void testGetStats() throws Exception {
         consumer.close();
         producer.close();
     }
+    @Test(timeOut = 100000)
+    public void testMessageListenerGetStats() throws Exception {
+        final String topicName = "persistent://my-property/my-ns/testGetStats" + UUID.randomUUID();
+        final String subName = "my-sub";
+        final int receiveQueueSize = 100;
+        @Cleanup
+        PulsarClient client = newPulsarClient(lookupUrl.toString(), 100);
+        Producer<String> producer = pulsarClient.newProducer(Schema.STRING)
+                .enableBatching(false).topic(topicName).create();
+        ConsumerImpl<String> consumer = (ConsumerImpl<String>) client.newConsumer(Schema.STRING)
+                .messageListener((MessageListener<String>) (consumer1, msg) -> {
+                    try {
+                        TimeUnit.SECONDS.sleep(10);
```

Comment:
Pulsar tests take a long time to run. This test is only verifying correct counting of a queue. Perhaps a faster solution might use futures that are completed at specific locations in the test to verify the behavior.

Reply:
I ran this test locally, and it is only 3 seconds. Since we're not reusing the client, this test is fine.